Source/ImPlotWidgets/Private/LineFilledGraphWidget.cpp

```diff
@@ -51,6 +45,18 @@ void SFilledGraphWidget::OnImGuiDraw() const
 	constexpr int32 GraphFlags = ImPlotFlags_AntiAliased;
 	constexpr int32 AxisFlags = ImPlotAxisFlags_NoGridLines;
 
+	TArray<float> VisibleLineMaxValues;
+
+	TArray<TArray<float>> FittedYValues;
+	for (int32 i = 0; i < YValueArrays.Num(); ++i)
```

Comment:
If I understand right, we run this unit conversion every frame, even if the unit doesn't change? Could we have some concept of whether the unit changed last frame and therefore we need to update the values this frame?

Reply:
Yeah so we had a chat about this with Sam and basically, the unit can only change if new lines are being hidden/displayed. The checkboxes return a bool which states if they've been pressed, and we should be able to use that to check whether we need to convert the units and that way we can leave out any unnecessary calculations

Reply:
See [this comment](https://github.com/CoconutLizard/UnrealImGui/pull/15#discussion_r768832517) and further comments relating to the same idea about calculating the best-fit values and caching them instead of every frame